.github/workflows/integration_test_run.yaml

```diff
@@ -252,6 +253,31 @@ jobs:
             module="-k ${{ matrix.modules }}"
           fi
           tox -e integration -- --model testing --keep-models $series $module $args ${{ inputs.extra-arguments }}
+      - name: Run lxd integration tests
+        working-directory: ${{ inputs.working-directory }}
+        if: ${{ inputs.provider == 'lxd' }}
+        run: |
+          echo "CHARM_NAME=$([ -f metadata.yaml ] && yq '.name' metadata.yaml || echo UNKNOWN)" >> $GITHUB_ENV
+          args=""
+          for image_name in $(echo '${{ inputs.images }}' | jq -cr '.[]'); do
+            if [ ${{ github.event.pull_request.head.repo.fork }} = "true" ]; then
+              args="${args} --${image_name}-image localhost:32000/${image_name}:latest"
+            else
+              args="${args} --${image_name}-image ${{ inputs.registry }}/${{ inputs.owner }}/${image_name}:${{ github.run_id }}"
+            fi
+          done
+
+          series=""
+          if [ ! -z ${{ matrix.series }} ]; then
+            series="--series ${{ matrix.series }}"
+          fi
+          module=""
+          if [ ! -z ${{ matrix.modules }} ]; then
+            module="-k ${{ matrix.modules }}"
+          fi
+          tox -e integration -- --keep-models $series $module $args ${{ inputs.extra-arguments }}
```

Comment:
Isn't there a way to do a bit of factoring between this step and the previous one as 90% of the run script is similar ?
Like having a models_arg output for both steps
And a new step that does all the args/module/series and executes tox using the output of the previous step

Reply:
Not that I know of, but I'll take a look and ask around

Reply:
I can do that as part of #145 if that works for you. I'll have to resolve the conflict that will arise as part of this anyway